pkg/controller/config/config_controller.go

```diff
@@ -150,13 +151,25 @@ func (r *ReconcileConfig) Reconcile(ctx context.Context, request reconcile.Reque
 	}
 	exists := true
 	instance := &configv1alpha1.Config{}
+	cfgTracker := r.getTracker()
+
 	err := r.reader.Get(ctx, request.NamespacedName, instance)
 	if err != nil {
+		// use a tempInstance for cfg tracker
+		tempInstance := &configv1alpha1.Config{}
+		tempInstance.SetNamespace(request.NamespacedName.Namespace)
+		tempInstance.SetName(request.NamespacedName.Name)
+
 		// if config is not found, we should remove cached data
 		if errors.IsNotFound(err) {
 			exists = false
+			// expectations will be reaped by the tracker for deleted resources.
```

Comment:
is this true?  Can't remember seeing this.

Reply:
chatted offline that this does indeed happen here:

https://github.com/open-policy-agent/gatekeeper/blob/366ba791d009c70f18f85fd94dbd5aed8426fc9d/pkg/readiness/ready_tracker.go#L449